tests/test_queries.py

```diff
@@ -0,0 +1,176 @@
+        returned_message_ids = {msg["id"] for msg in messages}
+
+        for message in MTMessage.objects.all():
+            if message.id in returned_message_ids:
+                assert message.status == MTMessage.Status.SENDING
+            else:
+                assert message.status == MTMessage.Status.NEW
+
+
+@pytest.mark.django_db
+class TestGetMessagesToProcess(object):
+    def test_empty(self):
+        """Return an empty list if there are no queued inbound messages."""
+        messages = get_mo_messages_to_process(limit=1)
+        assert messages.count() == 0
+
+    def test_partial_page(self):
+        """Return fewer than `limit` messages if appropriate."""
+        MOMessageFactory.create_batch(5)
+
+        messages = get_mo_messages_to_process(limit=100)
+
+        assert messages.count() == 5
+
+    def test_full_page(self):
+        """Return only `limit` messages if more are present."""
+        MOMessageFactory.create_batch(5)
+
+        messages = get_mo_messages_to_process(limit=3)
+
+        assert messages.count() == 3
+
+    def test_new_messages_only(self):
+        """Return only messages with status NEW."""
+        backend = BackendFactory()
+        new_messages = [MOMessageFactory(backend=backend, status=MOMessage.Status.NEW)]
+        for other_status in [
+            val for val in MOMessage.Status.values if val != MOMessage.Status.NEW
+        ]:
+            MOMessageFactory.create_batch(
+                3,
+                backend=backend,
+                status=other_status,
+            )
+        new_messages.append(
+            MOMessageFactory(backend=backend, status=MOMessage.Status.NEW)
+        )
+
+        messages = get_mo_messages_to_process(limit=10)
+
+        assert len(messages) == 2
+        assert set(messages) == set(new_messages)
+
+    def test_updates_status_to_processing(self):
+        """Returned messages should be updated to status PROCESSING."""
+        backend = BackendFactory()
+        MOMessageFactory.create_batch(5, backend=backend)
+
+        messages = get_mo_messages_to_process(limit=3)
+        returned_message_pks = {msg.pk for msg in messages}
+
+        for message in MOMessage.objects.all():
+            if message.pk in returned_message_pks:
+                assert message.status == MOMessage.Status.PROCESSING
+            else:
+                assert message.status == MOMessage.Status.NEW
+
+
+@pytest.mark.django_db(transaction=True)
+class TestNotifications(object):
+    def test_listen_notify_new_messages(self):
+        """`pg_notify` should publish messages that can be read by the
+        connection made from `pg_listen`.
+        """
+        listen_conn = pg_listen("test_channel")
+
+        listen_conn.poll()
+        assert len(listen_conn.notifies) == 0
+
+        for _ in range(5):
+            pg_notify("test_channel")
+
+        listen_conn.poll()
+        assert len(listen_conn.notifies) == 5
+
+        drain_conn(listen_conn)
```

Comment:
I understand the theory of `drain_conn()`, but I couldn't get it to break by removing. In what situation does it break?